### ISSUE TITLE
useTransition: "enterTo" remains during "leave" (and vice versa on "enter")

### ISSUE THREAD

**Issue by weaverryan** [stimulus-use/stimulus-use#110]
Hi there!

Thanks for the wonderful library - i'm building a tutorial with that uses it right now - it's awesome!

I'm testing out the beta transitions. I'm not using Tailwind in this app, so the enter-leave classes are not the same as the sandbox. I've noticed a slight issue. For example, suppose we have:

```js
    connect() {
        useClickOutside(this);
        useDebounce(this);
        useTransition(this, {
            element: this.resultTarget,
            enter: 'transition-fade-enter',
            enterActive: 'transition-fade-enter-active',
            enterTo: 'transition-fade-enter-to',
            leave: 'transition-fade-leave',
            leaveActive: 'transition-fade-leave-active',
            leaveTo: 'transition-fade-leave-active-to',
        });
    }
```

The `enter()` works great, the element looks like this after:

```html
<div
    class="transition-fade-enter-to"
    data-search-preview-target="result"
> ....</div>
```

However, when I `leave()`, the `enterTo` class (which is `transition-fade-enter-to`) *remains* on the element permanently. For example, once `leave()` finishes, the HTML looks like this:

```html
<div
    class="transition-fade-enter-to transition-fade-leave-active-to hidden"
    data-search-preview-target="result"
> ....</div>
```

Basically, both the `enterTo` and `leaveTo` classes are "stuck" on there permanently. This kills all future transitions because, for example,  if `transition-fade-enter-to` has `opacity: 1` that overrides the `opacity 0` set by `transition-fade-leave-active-to`. 

I think the solution is "simply" to, on leave, remove the `enterTo` class if it's there. And on enter, remove the `leaveTo` class if it is there.

Let me know if that makes sense :).

Cheers!

**Comment by weaverryan** [comment]
Btw, I tried my luck on #111 with a PR ;) 

Cheers!

**Comment by adrienpoly** [comment]
Thanks I have seen something similar on my end recently. My plan is to finalize the edge case for useTransition next week and officially release the documentation. Will look at your PR at that time thanks 🙏 .

**Comment by weaverryan** [comment]
Thanks @adrienpoly - that's great news! I'm hoping to include it in my tutorial (releasing now)... but I'll keep it to a final chapter to hopefully buy some extra time. Keep up the good work!

**Comment by adrienpoly** [comment]
@all-contributors please add @weaverryan to bug
